### PR TITLE
Junit testing

### PR DESCRIPTION
**Pull request by peterbae**
Added CallableStatementTest and PrecisionScaleTest

### CODE REVIEW

File: src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/CallableStatementTest.java [hunk 1]
```diff
@@ -0,0 +1,2645 @@
+package com.microsoft.sqlserver.jdbc.AlwaysEncrypted;
```
Comment (v-afrafi) [inline src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/CallableStatementTest.java]: missing header

File: src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/PrecisionScaleTest.java [hunk 2]
```diff
@@ -0,0 +1,578 @@
...
+
+        createDatePrecisionTable(0);
+        populateDateNormalCase(0);
+        populateDateSetObject(0);
+
+        testDate(dateNormalCase2, dateSetObject2);
+    }
+
+    @Test
+    public void testNumericPrecision8Scale2_Null() throws Exception {
```
Comment (v-afrafi) [inline src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/PrecisionScaleTest.java]: remove _ in the test name

File: src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/PrecisionScaleTest.java [hunk 3]
```diff
@@ -0,0 +1,578 @@
...
+        // Cannot use date.toGMTString() here directly since the date object is used elsewhere for population of data.
+        GMTDate = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss").format(date);
+        GMTDateWithoutDate = new SimpleDateFormat("HH:mm:ss").format(date);
+
+        // datetimeoffset is aware of timezone as well as Date, so we must apply the timezone value twice.
+        dateTimeOffsetExpectedValue = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss")
+                .format(new Date(1450812362177L - offsetFromGMT - offsetFromGMT + offset));
+    }
+
+    @Test
```
Comment (v-afrafi) [inline src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/PrecisionScaleTest.java]: Great work. One last comment is to add javadocs for the test methods and a general description at the beginning of the file explaining what each class file is containing and testing.

File: src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/CallableStatementTest.java (hunk 1 above)
Comment (peterbae) [inline src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/CallableStatementTest.java]: done.

File: src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/CallableStatementTest.java [hunk 4]
```diff
@@ -0,0 +1,2645 @@
...
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.platform.runner.JUnitPlatform;
+import org.junit.runner.RunWith;
+
+import java.math.BigDecimal;
+import java.sql.Date;
+import java.sql.DriverManager;
```
Comment (peterbae) [inline src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/CallableStatementTest.java]: done.

File: src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/CallableStatementTest.java [hunk 5]
```diff
@@ -0,0 +1,2645 @@
...
+        testOutputProcedureBatchInorder("{call " + outputProcedureBatch + "(?,?,?,?)}");
+    }
+
+    @Test
+    public void testOutputProcedure4() throws SQLException {
+        createOutputProcedure4();
+    }
+
+    private static void dropTables() throws SQLException {
+        stmt.executeUpdate("if object_id('" + table1 + "','U') is not null" + " drop table " + table1);
```
Comment (peterbae) [inline src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/CallableStatementTest.java]: done.

File: src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/PrecisionScaleTest.java (hunk 3 above)
Comment (peterbae) [inline src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/PrecisionScaleTest.java]: done.

File: src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/PrecisionScaleTest.java (hunk 2 above)
Comment (peterbae) [inline src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/PrecisionScaleTest.java]: done.

**Comment by v-afrafi** [comment]
Thank you @peterbae for fixing other test classes!

**Pull request merged**